src/CHIPGraph.cc

```diff
@@ -150,21 +173,18 @@ CHIPGraphNodeKernel::CHIPGraphNodeKernel(const void *HostFunction, dim3 GridDim,
   Params_.gridDim = GridDim;
   Params_.kernelParams = Args;
   Params_.sharedMemBytes = SharedMem;
-
   auto Dev = Backend->getActiveDevice();
-  CHIPKernel *ChipKernel = Dev->findKernel(HostPtr(HostFunction));
-  if (!ChipKernel)
+  Kernel_ = Dev->findKernel(HostPtr(Params_.func));
+  if (!Kernel_)
     CHIPERR_LOG_AND_THROW("Could not find requested kernel",
                           hipErrorInvalidDeviceFunction);
-  ExecItem_ =
-      Backend->createCHIPExecItem(GridDim, BlockDim, SharedMem, nullptr);
-  ExecItem_->setKernel(ChipKernel);
-
-  ExecItem_->copyArgs(Args);
-  ExecItem_->setupAllArgs();
+  ExecItem_ = Backend->createCHIPExecItem(Params_.gridDim, Params_.blockDim,
+                                          Params_.sharedMemBytes, nullptr);
+  ExecItem_->setKernel(Kernel_);
 }
 
-int NodeCounter = 1;
+static unsigned NodeCounter = 1;
```

Comment:
bit of a random placement and no explanation

Reply:
I see it makes the variable not exported which is probably the intent. That said, I think the variable should be atomic to avoid race condition in CHIPGraph::addNode().